### DISCUSSION TITLE
Why do my switches have dimmer related settings?

### DISCUSSION THREAD

**Discussion by jmwhite5** [zwave-js/zwave-js-ui#3724] [Q&A] [answered]
**Disclaimer**: I am in the middle of trying to investigate why my z-wave network melts down occasionally.  I have noticed that when the system is refreshing values from nodes, several of my simple switches are polled for 255 dimmingDuration values.  It looks like this:

```
2024-05-26 10:10:40.570 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-247 => NaNs
2024-05-26 10:10:40.571 INFO Z-WAVE: [Node 030] Value added 30-44-0-level-248 => undefined
2024-05-26 10:10:40.572 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-248 => NaNs
2024-05-26 10:10:40.573 INFO Z-WAVE: [Node 030] Value added 30-44-0-level-249 => undefined
2024-05-26 10:10:40.574 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-249 => NaNs
2024-05-26 10:10:40.574 INFO Z-WAVE: [Node 030] Value added 30-44-0-level-250 => undefined
2024-05-26 10:10:40.575 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-250 => NaNs
2024-05-26 10:10:40.576 INFO Z-WAVE: [Node 030] Value added 30-44-0-level-251 => undefined
2024-05-26 10:10:40.577 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-251 => NaNs
2024-05-26 10:10:40.578 INFO Z-WAVE: [Node 030] Value added 30-44-0-level-252 => undefined
2024-05-26 10:10:40.579 INFO Z-WAVE: [Node 030] Value added 30-44-0-dimmingDuration-252 => NaNs
```

This causes a storm of z-wave traffic 255 zwave requests x dozens of devices.  My zwave controller ends up freezing up and I have to do a soft reset to recover.  

**So my question is**, why are my switches trying to retrieve irrelevant data (dimmer settings for lights) 

Additional context: 
I am using Home Assistant and some of my switches I have them show as lights.  Maybe HA is causing node z-wave JS to poll for light related settings?

But I have a counter example of a switch that also have this issue: (Enbrighten model [55256](https://devices.zwave-js.io/?jumpTo=0x0063:0x4952:0x3234:0.0))

<img width="1210" alt="Screenshot 2024-05-26 at 11 15 39" src="https://github.com/zwave-js/zwave-js-ui/assets/1237042/cf07b409-4513-4ef1-81ac-9f5b049ddc30">

**Comment by kpine** [comment]
Those particular log messages don't necessarily indicate anything on the Z-Wave network was polled. You'd need to look at driver debug logs to confirm that. https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs

In fact, the "undefined" and "NaNs" values mean they were never polled. This command class is not polled during interview, you need to do it manually.

**Reply by jmwhite5** [reply]
> Those particular log messages don't necessarily indicate anything on the Z-Wave network was polled. You'd need to look at driver debug logs to confirm that. https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs
> 
> In fact, the "undefined" and "NaNs" values mean they were never polled. This command class is not polled during interview, you need to do it manually.

Thanks.  I think you're right.  I happened to have driver logs enabled ([zwavejs_2024-05-26.log](https://github.com/zwave-js/zwave-js-ui/files/15448728/zwavejs_2024-05-26.log))  and looking into them, I don't see any dimmerDuration values being polled.  

So probably unrelated to my controller jamming issue.

But I guess I'm still curious why those values are there anyways.  I've turned on my UI logs and will provide them when I repro.

**Reply by kpine** [reply]
The "value added" messages are related to the [Values API](https://zwave-js.github.io/node-zwave-js/#/api/valueid), which helps ZUI display them in the control panel. You'll see "value added" for every node and existing value whenever ZUI restarts, or you add a new node. "value updated" would typically be related to actual device activity (or optimistic driver settings).

As you noticed, there are 255 x 2 values for each node for Scene Actuator Configuration. Obviously, it would take forever to read the initial values of these when a notice is interviewed, so for this case only the metadata is created. The existence of the values means ZUI can display them and you can set them.

**Reply by jmwhite5** [reply]
> The "value added" messages are related to the [Values API](https://zwave-js.github.io/node-zwave-js/#/api/valueid), which helps ZUI display them in the control panel. You'll see "value added" for every node and existing value whenever ZUI restarts, or you add a new node. "value updated" would typically be related to actual device activity (or optimistic driver settings).
> 
> As you noticed, there are 255 x 2 values for each node for Scene Actuator Configuration. Obviously, it would take forever to read the initial values of these when a notice is interviewed, so for this case only the metadata is created. The existence of the values means ZUI can display them and you can set them.

I see.  And what would cause a switch to allow dimmerDuration values to be added to the UI?  Does it just do that for any device type just in case it's supported by the device?  Or is there a schema somewhere that maps a device type to a set of allowable values?  (and where is that schema? and is the schema broken? or wrong?)

**Reply by kpine** [reply, marked as answer]
Your device reports that it supports Scene Actuator Configuration CC. That CC is used for configuring  "native" Z-Wave scenes, which is a rarely used feature (from what I can tell). The specification explains the details. https://github.com/zwave-js/specs/blob/master/Z-Wave%20Specification%20AWG%20V3.0.pdf

**Reply by jmwhite5** [reply]
Thank you @kpine, the document mentions that a device without the capability must ignore this field.  So even though it's confusing at first glance, I think it's operating as designed.

```
Dimming Duration (8 bits)
This field MUST specify the time it MUST take to reach the target level associated to the actual Scene ID.
A receiving node always starts from current level. The dimming duration MUST be the same inde- pendently of the number of levels to be changed.
Supporting actuator nodes without duration capabilities MUST ignore this field.
Supporting actuator nodes with duration capabilities SHOULD respect the indicated duration to reach the target level.
This field MUST be encoded according to Table 2.9.
```
